common/src/reserved.rs

```diff
@@ -338,4 +408,231 @@ mod tests {
                 .collect::<HashSet<_>>()
         );
     }
+
+    #[test]
+    fn test_apply_delegate_on_governance_success() {
+        // given
+        setup_test();
+        let (mut state, keys) = generate_standard_genesis(3);
+
+        // delegator : member-000 && delegatee : member-002
+        let delegator_public_key = keys[0].0.clone();
+        let delegator_private_key = keys[0].1.clone();
+        let delegatee_public_key = keys[2].0.clone();
+
+        let delegatee = state.members[2].clone();
+
+        // when
+        let data: DelegationTransactionData = DelegationTransactionData {
+            delegator: delegator_public_key,
+            delegatee: delegatee_public_key,
+            governance: true,
+            block_height: 0u64,
+            timestamp: 0i64,
+        };
+        let proof = TypedSignature::sign(&data, &delegator_private_key).unwrap();
+
+        let tx = TxDelegate { data, proof };
+        let new_state = state.apply_delegate(&tx);
+
+        // then
+        assert!(new_state.is_ok());
+        let new_state = new_state.unwrap();
+        let new_state_validator_set = new_state.get_validator_set();
+        let new_state_governance_set = new_state.get_governance_set();
+
+        assert_eq!(
+            new_state.members[0].governance_delegatee.as_ref().unwrap(),
+            &delegatee.name
+        );
+
+        assert_eq!(
+            new_state.members[0].consensus_delegatee.as_ref().unwrap(),
+            &delegatee.name
+        );
+
+        assert_eq!(
+            new_state_validator_set
+                .unwrap()
+                .into_iter()
+                .find(|v| v.0 == delegatee.public_key)
+                .unwrap()
+                .1,
+            2
+        );
+
+        assert_eq!(
+            new_state_governance_set
+                .unwrap()
+                .into_iter()
+                .find(|v| v.0 == delegatee.public_key)
+                .unwrap()
+                .1,
+            2
+        );
```

Comment:
I was thinking of comparing the validator set and governance set that can be obtained by `generate_delegated_genesis` but this seems okay too. 

@junha1 how do you think about it?